Run JetBrains PyCharm code formatter on source and fix several PEP8 lint messages

urzds opened the pull request:
This will make it easier to spot the errors and warnings PyCharm issues during static analysis of the code.

metasoarous commented on tripl/nestly.py:
```diff
@@ -401,53 +413,53 @@ def add_parent_attrs(parent_nest_level):
-                #print "Finding attr in parent nest from main loop", attr
-                #c[attr].append(translated_metadata)
+            # for attr in parent_nest['aggregate_attrs']:
+            # print "Finding attr in parent nest from main loop", attr
```
This identation is not right.

metasoarous commented on tripl/nestly.py:
```diff
@@ -460,10 +472,8 @@ def _metadata(outdir, c):
+    # def add_aggregate(self, name, data_fac):
+    # return self.scons_wrap.add_aggregate(name, data_fac)
     # For backwards compatibilty; eventually
-    #def add_controls(self, name, data_fac):
```
Again, do not support nuking code indentation in comments.

metasoarous commented on tripl/tripl.py:
```diff
@@ -72,7 +76,7 @@ def __iter__(self):
                     yield k, v
             else:
-                for tupl in v:
+                for tupl in vs:
```
Did this find a bug?

urzds replied:
PyCharm said `v` might no be assigned before use. `vs` seemed to be the closest to what might have been meant.

metasoarous replied:
I think you're right that this is the only thing that seems to make sense. Will look into it further. Thanks again!

metasoarous commented on tripl/tripl.py:
```diff
@@ -541,37 +546,36 @@ def loads(cls, filenames, schema=None, id_attrs=None):
 
     # My python Datalog grammar proposal:
-    ##base query; matches graph
-    #{'find': ['?x', '?y'],
```
Please un-nuke this indentation

urzds replied:
Sorry, it appears PyCharm had trouble unindenting `  #x` into `#  x`.

metasoarous reviewed (commented):
Thanks for working on this! This mostly looks good, but there are a few places where I think it made some overzealous changes.

urzds commented on tripl/tripl.py:
```diff
@@ -29,29 +31,32 @@ def log(name, value):
-    "return some thing from the set, or None if nothing"
-    if isinstance(xs, (str, unicode, int, float, bool, dict, Entity, uuid.UUID)):
+    """return some thing from the set, or None if nothing"""
+    if isinstance(xs, (str, int, float, bool, dict, Entity, uuid.UUID)):
```
Please note that I removed `unicode` from the list, because that does not exist in Python 3. This might break Python 2.

metasoarous replied:
Hmm... this may be a problem. I'll look into it. Stupid that a `unicode` isn't also a `str`. Thanks.

urzds replied:
In Python 2 `unicode` is for strings and `str` for binary data. In Python 3 `str` is for strings and `bytes` for binary data.

urzds commented on tripl/tripl.py:
```diff
@@ -180,13 +184,15 @@ def __getitem__(self, key):
+        try:
+            return self[key]
+        except KeyError:
+            return default
```
PyCharm noticed that `default` was never used. I changed functionality to what was probably intended.

metasoarous replied:
I think you're right about that.

urzds commented on tripl/tripl.py:
```diff
@@ -583,27 +588,29 @@ def _entity_lookup(self, lookup):
+
         pattern = {k: xf_subpattern(v) for k, v in pattern.items()}
-        return reduce(set.intersection, map(self._entity_lookup, pattern.items()))
+        return functools.reduce(set.intersection, map(self._entity_lookup, pattern.items()))
```
PyCharm notices a type mismatch here (it expected a function `(T, T) -> T` but got something that goes from `T` to a set of `T`), but I was unable to figure out whether this is a false positive or if it is a bug, how it should actually be.

urzds replied:
The warning can be silenced by replacing the line with this:
```python
        def intersection(a, b):
            return a.intersection(b)

        return functools.reduce(intersection, map(self._entity_lookup, pattern.items()))
```

Could it be that `set.intersection` cannot / should not be used as a static method in Python 3 anymore or is this a false alarm?

metasoarous replied:
I think it's probably a false alarm (PyCharm outsmarting itself). IMHO, it's one of the nicer features of pythons that methods are just functions, and that you can do this sort of functional thing when it suits.

urzds commented on tripl/tripl.py:
```diff
@@ -667,12 +674,13 @@ def pull(self, pull_expr, entity,
             for dict_pattern in dict_patterns:
                 for attr, token in dict_pattern.items():
                     reverse = reverse_lookup(attr)
+                    eids = None
```
PyCharm noticed that `eids` might be undefined below, so I explicitly set it to `None` here.

urzds commented on tripl/tripl.py:
```diff
@@ -715,27 +724,26 @@ def pull_many(self, pull_expr, eids_or_pattern, sort_by=None, sort_desc=True):
+            results = sorted(results, key=lambda x: x[sort_by])
         if not sort_desc:
-            results = results.reversed()
+            results = reversed(results)
```
This might be a Python 3 change: `results` has no field `reversed`, so I changed this to use the `reversed` builtin function instead.

urzds commented on tripl/data/toy.csv:
```diff
@@ -1,5 +1,5 @@
+EBOV ,jena   ,s1    ,i1,2017-06-01,t1
+EBOV ,jena   ,s2    ,i2,2017-06-01,t1
+EBOV ,seattle,s3    ,i3,2017-06-03,t2
+IAV  ,seattle,s3    ,i4,2017-06-03,t2
```
This was unintentional. Probably caused by a data file appearing in the source directory that I reformatted in order to squelch PEP8 warnings. I can revert, if you want.

metasoarous replied:
Yes, please. Thanks.

urzds commented on tripl/nestly.py:
```diff
@@ -16,7 +14,10 @@
 import bio
 
-#import tripl
+# pp  # lint
```
This line was active before, but I could not figure out what the expression `pp` is supposed to do. PyCharm suggested it would have no effect at all.

metasoarous replied:
The reason I did that is that I get warnings in Vim when I import a module and then don't use it. So here I was just "using it" as a no-op so that the vim linter doesn't complain any more.

Commit:
Run JetBrains PyCharm code formatter on source and fix several PEP8 code formatting messages

This will make it easier to spot the errors and warnings PyCharm issues during static analysis of the code.

ghost commented:
I split out all non-formatting related changes into separate branches on top of this one, in order to make review easier.

metasoarous commented:
@urzds; Great! Thanks so much for separating out the more substantial changes. Merging now.
